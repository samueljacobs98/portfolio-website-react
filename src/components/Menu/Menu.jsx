import "./Menu.scss";
import hamburger from "../../assets/svgs/hamburger.svg";
import closeBtn from "../../assets/svgs/close-button.svg";
import logo from "../../assets/svgs/logo.svg";
import { useState } from "react";
import Links from "../Links/Links";

const Menu = () => {
  const [showMenu, setShowMenu] = useState(false);

  const toggleMenu = () => {
    setShowMenu(!showMenu);
  };

  return (
    <>
      {!showMenu && (
        <button onClick={toggleMenu}>
          <img className="hamburger" src={hamburger} alt="Open menu" />
        </button>
      )}
      {showMenu && (
        <>
          <menu className={showMenu ? "menu" : "menu menu--inactive"}>
            <img
              className="menu__close"
              src={closeBtn}
              alt="Open menu"
              onClick={toggleMenu}
            />
            <img
              className="menu__logo"
              src={logo}
              alt="Open menu"
              onClick={toggleMenu}
            />
            <ul className="menu__links">
              <li>
                <a className="menu__link" href="#projects" onClick={toggleMenu}>
                  Projects
                </a>
              </li>
              <li>
                <a className="menu__link" href="#about" onClick={toggleMenu}>
                  About
                </a>
              </li>
              <li>
                <a className="menu__link" href="#contact" onClick={toggleMenu}>
                  Contact
                </a>
              </li>
            </ul>
            <Links />
          </menu>
        </>
      )}
    </>
  );
};

export default Menu;
