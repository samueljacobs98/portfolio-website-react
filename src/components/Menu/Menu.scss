@use "../../assets/scss/variables.scss" as *;

.menu {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  z-index: 1;
  background-color: $colour-background;

  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;

  &__links {
    list-style-type: none;
    text-align: center;
    height: 30%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
  }

  &__link,
  &__link:visited {
    text-decoration: none;
    color: $color-white;
    font-size: $font-size;
  }

  &--inactive {
    background-color: transparent;
  }

  .links {
    display: flex;
    position: relative;
  }
}

.hamburger,
.menu__close {
  position: fixed;
  top: 0;
  right: 0;
  margin: $margin-fixed;
  z-index: 1;

  &:hover {
    cursor: pointer;
  }
}
