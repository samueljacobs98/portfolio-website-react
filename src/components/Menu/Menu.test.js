import { render } from "@testing-library/react";
import Menu from "./Menu";

test("should match snapshot", () => {
  const { container } = render(<Menu />);
  expect(container).toMatchSnapshot();
});
