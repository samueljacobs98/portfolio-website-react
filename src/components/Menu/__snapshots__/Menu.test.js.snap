// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <button>
    <img
      alt="Open menu"
      class="hamburger"
      src="hamburger.svg"
    />
  </button>
</div>
`;
